sympy/functions/elementary/piecewise.py

```diff
@@ -462,20 +465,22 @@ def _eval_subs(self, old, new):
     def _eval_transpose(self):
         return self.func(*[(e.transpose(), c) for e, c in self.args])
 
-    def _eval_template_is_attr(self, is_attr, when_multiple=None):
```

Comment:
One would only inspect the condition if the expression were a function of a common variable. And no, that is not included and cannot be until I get to the point of having the robust condition checking in place -- it is currently very easy to trip up the condition checking. Once that is in place, any condition sharing a single variable with the expression could be solved and substituted into the expression when doing the checking so Piecewise arg

```
(a + b - 3, a + b < 2)` -> solve cond for a and subst using positive symbol `pos` to indicate that we are using "<"
(2 - b - pos) + b - 3 ->
-1 - pos -> a negative
```
could be inferred.